src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java

```diff
@@ -512,7 +517,17 @@ protected Future<Void> recreateItemRecords(Map<String, List<Piece>> piecesGroupe
           var dstConfig = InventoryUtils.constructItemRecreateConfig(checkInPiece.getReceivingTenantId(), requestContext, false);
           if (InventoryUtils.allowItemRecreate(srcConfig.tenantId(), dstConfig.tenantId())) {
             logger.info("recreateItemRecords:: recreating item by id '{}', srcTenantId: '{}', dstTenantId: '{}'", piece.getItemId(), srcConfig.tenantId(), dstConfig.tenantId());
-            futures.add(itemRecreateInventoryService.recreateItemInDestinationTenant(itemToRecreate.getCompositePoLine(), piece, srcConfig.context(), dstConfig.context()));
+
+            // Fetch purchase order asynchronously
+            Future<CompositePurchaseOrder> poFuture = purchaseOrderStorageService.getPurchaseOrderByIdAsJson(itemToRecreate.getPoLineId(), requestContext)
```

Comment:
I suggest you change the Reference type and variable name to `var purchaseOrderFuture = ...`